collector/factory.go

```diff
@@ -0,0 +1,57 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package collector // import "go.opentelemetry.io/ebpf-profiler/collector"
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer/xconsumer"
+	"go.opentelemetry.io/collector/receiver"
+	"go.opentelemetry.io/collector/receiver/xreceiver"
+
+	"go.opentelemetry.io/ebpf-profiler/collector/internal"
+	"go.opentelemetry.io/ebpf-profiler/internal/controller"
+)
+
+var (
+	typeStr = component.MustNewType("otelreceiver")
```

Comment:
[xreceiver#NewFactory](https://pkg.go.dev/go.opentelemetry.io/collector/receiver/xreceiver#NewFactory) does not a good job in defining `cfgType`. So I might be wrong on this: To me `otelreceiver` is too generic as this `receiver.Factory` is focused on profiling data. Is `profiling_receiver` a better alternative?

Reply:
You're right, the current name is wrong.
Looking at other receivers, they don't include the term `receiver`. So I've named this `profiling`.